sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnAdapters.java

```diff
@@ -184,9 +293,17 @@ public void outputWithTimestamp(OutputT output, Instant timestamp) {
 
     @Override
     public BoundedWindow window() {
-      // The DoFn doesn't allow us to ask for these outside ProcessElements, so this
+      // The OldDoFn doesn't allow us to ask for these outside processElement, so this
       // should be unreachable.
-      throw new UnsupportedOperationException("Can only get the window in ProcessElements");
+      throw new UnsupportedOperationException("Can only get the window in processElement");
+    }
+
+    @Override
+    public WindowingInternals<InputT, OutputT> windowingInternals() {
+      // The OldDoFn doesn't allow us to ask for these outside ProcessElements, so this
+      // should be unreachable.
+      throw new UnsupportedOperationException(
+          "Can only get WindowingInternals in processElement");
```

Comment:
Nope, in fact we don't. `OldDoFn.Context` does not have `windowingInternals()`. It is only used in `ReduceFnRunner` (where it should be replaced by the already-created interface `OutputWindowedValue`) and that is in the context of `processElement`.